docs/scripts/api/buildApi.ts

```diff
@@ -78,29 +87,111 @@ export default async function linkifyTranslation(
   );
 }
 
+const specialAPIs: { folder: string; packages: XProjectNames[]; documentedInterfaces: string[] }[] =
```

Comment:
Would it make sense to define those in a new file next to `getComponentInfo` in `buildApiDocs`?

Reply:
- make sens: yes
- feasible: no

😅

The API build is done in two times in X docs:
```
yarn docs:api:build && yarn docs:api:buildX
```

1. The generation of components API done by the pipeline we import from core and defined by those files in `buildApiDocs`
2. The one specific to X that generate files for the events, selectors, interfaces and API

I tried once to merge those two scripts. But it's... complicated.

The core script is written without taking care about project relations, because each component can be done independently.
The X script is more project related, because knowing in which project properties are defined implies their pricing level (community, pro, premium) Plus when generating pages to document interfaces, we update an object to know all the pages generated, and then override the translation such that `[[GridAPI]]` is transformed in a link to the API page of the grid api.


In brief, core does
```py
for project in projects:
  for component in project:
	generateApiPage(component)
```

And X does
```py
generatedPages = {}
for interface in interfacesToDocument:
  generateApiPage(interface, projects, generatedPages)

for project in dataGridProjects:
  for apiInterface in project:
	generateApiPage(apiInterface)

for translation in treanslationFiles:
  updateLinks(translation, generatedPages)
```

Reply:
> The more we centralize, the better I am 😆

That's a [political opinion](https://www.collectivites-locales.gouv.fr/institutions/historique-de-la-decentralisation)? 😇😆

> Then maybe moving it to `createXTypeScriptProjects`

The file name is not well adapt, but it will make those parameters more visible :+1: